### PR TITLE
Add 0.9.1 to 0.10.0 upgrade guide to docs

### LINKED ISSUE #3881: [0.10.0] Add 0.9.1 to 0.10.0 upgrade guide to docs
Part of release epic #3849.

As with previous releases, we should add an upgrade guide so that folks looking for instructions, e.g., regarding kernel troubleshooting and Tails instructions, can find them in a predictable location, not just in the blog.

### PR DESCRIPTION
**Pull request by edenemmanuel**
## Status

Ready for review

## Description of Changes

Fixes #3881.

SecureDrop release 0.10.0 is coming soon! This will add some documentation on how to upgrade from 0.9.1 to 0.10.0. In particular, we stress that the 3.14.x kernel will be removed in release 0.11.0.

## Testing

Reviewing for typos and running `make docs-lint` locally should suffice.

## Checklist

- [x] Doc linting (`make docs-lint`) passed locally

### CODE REVIEW

File: docs/upgrade/0.9.1_to_0.10.0.rst [hunk 1]
```diff
@@ -0,0 +1,50 @@
+Upgrade from 0.9.1 to 0.10.0
+============================
+
+Updating the Tails Workstations
+-------------------------------
+
+We recommend that you update all Tails drives to version 3.10, which was released
+concurrently with SecureDrop 0.9.0 on October 23, 2018. Follow the Tails
```
Comment (eloquence) [inline docs/upgrade/0.9.1_to_0.10.0.rst]: 0.9.0 -> 0.10.0

File: docs/upgrade/0.9.1_to_0.10.0.rst [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+graphical prompts on your workstations to perform this upgrade.
+
+Due to a bug in the graphical SecureDrop updater that was fixed in SecureDrop
+0.9.1 (released on September 6, 2018), attempting an update of your SecureDrop
+workstation code on your *Journalist* or *Admin Workstations* using the
+graphical updater will fail with an error message: "WARNING: Signature
+verification failed."
```
Comment (eloquence) [inline docs/upgrade/0.9.1_to_0.10.0.rst]: will fail -> may fail

I think there's potential for more refactoring here for clarity -- my personal bias would be to lead with the normal procedure (how to use the graphical updater), and then add the caveats (will not appear if running workstation code older than 0.7.0, may fail with error if running older than 0.9.1) & point to the docs as you do.

**Comment by eloquence** [comment]
Looking pretty good from my end, see comments above for small suggested edits. Thanks for preparing!

File: docs/upgrade/0.9.1_to_0.10.0.rst (hunk 1 above)
Comment (edenemmanuel) [reply docs/upgrade/0.9.1_to_0.10.0.rst]: Ack. Good one.

File: docs/upgrade/0.9.1_to_0.10.0.rst (hunk 2 above)
Comment (edenemmanuel) [reply docs/upgrade/0.9.1_to_0.10.0.rst]: Ack. I'll reach out for potential messaging in a DM @eloquence so as to avoid lots of commits.

**Comment by eloquence** [comment]
LGTM. Adding @redshiftzero for final sanity check. If there's one element that I still think could cause confusion, it's that a user might wonder whether following the 0.8.0 to 0.9.1 guide will take them all the way to 0.10.0 or not. It may be worth explicitly saying that, once the user has performed all the steps in that guide, they will be running 0.10.0 on the workstations, and no additional steps are required.

**Comment by edenemmanuel** [comment]
Thanks, y'all! :innocent:

**Comment by redshiftzero** [comment]
Ah right, so the admin tests are failing here due to:
 1. 0.10.0 tag being live but 
 2. 0.10.0 release object not being present yet

this is an inconsistent state that only occurs in this release period. Making release object now...

File: docs/upgrade/0.9.1_to_0.10.0.rst [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    git checkout 0.10.0
+
+.. important:: If you do see the warning "refname '0.10.0' is ambiguous" in the
+  output, we recommend that you contact us immediately at securedrop@freedom.press
+  (`GPG encrypted <https://securedrop.org/sites/default/files/fpf-email.asc>`__).
+
+Finally, run the following commands: ::
+
+  ./securedrop-admin setup
+  ./securedrop-admin tails-config
```
Comment (eloquence) [inline docs/upgrade/0.9.1_to_0.10.0.rst]: tails-config -> tailsconfig

Comment (edenemmanuel) [reply docs/upgrade/0.9.1_to_0.10.0.rst]: Ack. Fixing.

File: docs/upgrade/0.9.1_to_0.10.0.rst [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+workstations. Tails upgrades must be performed separately.
+
+If you don't see a graphical updater, you may be running an older
+version of the SecureDrop code on your workstation (earlier than
+0.7.0). You can update as follows: ::
+
+    cd ~/Persistent/securedrop
+    git fetch --tags
+    gpg --recv-key "2224 5C81 E3BA EB41 38B3 6061 310F 5612 00F4 AD77"
+    git tag -V 0.10.0
```
Comment (redshiftzero) [inline docs/upgrade/0.9.1_to_0.10.0.rst]: FYI this should be a lower case `-v`

Comment (edenemmanuel) [reply docs/upgrade/0.9.1_to_0.10.0.rst]: oy. fixing!

**Comment by redshiftzero** [comment]
other than these nits this looks good now, thanks @eloquence for noticing inconsistency with the release notes

**Comment by redshiftzero** [comment]
(btw admin tests are passing now that the release object is made :-))

**Comment by edenemmanuel** [comment]
Updated!

**Pull request merged**